### PR TITLE
Maintainability cleanup for EventSelector

### PR DESCRIPTION
**Pull request by wmtan**
This PR is a cleanup of the EventSelector class for maintainability and clarity. The changes include:
1) Removing one unused and one unnecessary data member.
2) Renaming some data members, function members, and function parameters to be more mnemonic
3) The three data members that are never changed are made const (more for clarity than for const correctness) and initialized in the initializer lists of the class. They are no longer modified in init() (renamed to initPathNames()), because they were just being reset to the same values.
4) The three const data members are the first three data members.
5) Iterator loops are changed to range for loops.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Framework/src/EventSelector.cc [hunk 1]
```diff
@@ -409,41 +389,39 @@ namespace edm
...
-  EventSelector::acceptOneBit(Bits const& b, 
-    		       	       HLTGlobalStatus const& tr, 
+  bool
+  EventSelector::acceptOneBit(Bits const& b,
+    		       	       HLTGlobalStatus const& tr,
     		               hlt::HLTState const& s) const
   {
     bool lookForException = (s == hlt::Exception);
     Bits::const_iterator i(b.begin());
     Bits::const_iterator e(b.end());
```
Comment (Dr15Jones) [inline FWCore/Framework/src/EventSelector.cc]: why were these variables kept?

Comment (wmtan) [inline FWCore/Framework/src/EventSelector.cc]: Chris,
    Just an oversight. I will remove them when you let me know that you have no other suggested changes, or in a half hour or so if I don't hear from you.

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**